controllers/news-controllers.js

```diff
@@ -0,0 +1,29 @@
+const { fetchAllTopics, fetchArticleById} = require("../models/news-models")
+
+exports.getAllTopics = (request, response, next) => {
+    fetchAllTopics()
+    .then((topics) => {
+        response.status(200).send({ topics })
+    })
+    .catch((error) => {
+        next(error)
+    })
+
+}
+exports.getArticlesById = (request, response) => {
+    const { article_id } = request.params;
+    fetchArticleById(article_id)
+    .then((articleData) => {
+        if(!articleData) {
+            return response.status(404).send({msg: 'Article not found'})
```

Comment:
have a think about whether you need the return statement here. try running the code without the return, with just reponse.status(404).send({msg: "article not found"})